### PR TITLE
Record events with ophan

### PR DESCRIPTION
**Pull request by svillafe**
## Why are you doing this?

We set up an example test in order to test the abacus stack for support frontend. In order to send an Ophan event when we click the link, I had to replace the `href` attribute from the `ctaCircle` component and pass an `onClick` function instead.

### Variants:
* control
* variantA

[**Trello Card**](https://trello.com/c/LgEpY7Ok/559-send-a%2Fb-test-results-to-ophan)

### First event
picture 269

### Complete event
picture 271

### CODE REVIEW

File: assets/helpers/abtest.js [hunk 1]
```diff
@@ -143,8 +155,8 @@ function getParticipation(mvtId: number): Participations {
 
 export const init = () => {
 
-  const mvt = getMvtId();
-  let participation = getParticipation(mvt);
+  const mvt: number = getMvtId();
```
Comment (davidfurey) [inline assets/helpers/abtest.js]: Why do you need to make the type explicit here?  Does flow not do type inference?

File: assets/helpers/abtest.js [hunk 2]
```diff
@@ -37,17 +37,29 @@ type Test = {
...
 
+type OphanABEvent = {
+  variantName: string,
+  complete: boolean,
+  campaignCodes?: string[],
+};
+
+
+type OphanABPayload = {
+  [TestId]: OphanABEvent,
```
Comment (davidfurey) [inline assets/helpers/abtest.js]: What does [TestId] mean?

File: assets/helpers/abtest.js (hunk 1 above)
Comment (svillafe) [inline assets/helpers/abtest.js]: I made it explicit in order to be more readable for someone who is reading the code and doesn't know what Mvt is.

File: assets/helpers/abtest.js (hunk 2 above)
Comment (svillafe) [inline assets/helpers/abtest.js]: It means that the name of the attribute will be type `TestId` and its value will be `OphanABEvent`.

File: assets/helpers/abtest.js [hunk 3]
```diff
@@ -165,6 +177,26 @@ export const getVariantsAsString = (participation: Participations): string => {
...
+  complete?: boolean = false,
+  campaignCodes?: string[] = []): void => {
+
+  const payload: OphanABPayload = {};
+
+  payload[testId] = {
+    variantName: variant,
+    complete,
+    campaignCodes,
+  };
```
Comment (Ap0c) [inline assets/helpers/abtest.js]: Possible shorted alternative, debatable whether it's clearer or not:

```js
const payload: OphanABPayload = {
  [testId]: {
    variantName: variant,
    complete,
    campaignCodes,
  },
};
```

File: assets/pages/bundles-landing/bundlesLanding.jsx [hunk 4]
```diff
@@ -25,6 +25,7 @@ import reducer from './reducers/reducers';
 
 const participation = abTest.init();
 
+
```
Comment (Ap0c) [inline assets/pages/bundles-landing/bundlesLanding.jsx]: Excellent adherence to Jamie's linting rules here, I like it 👍 (I suspect @davidfurey isn't so keen).

File: assets/helpers/abtest.js (hunk 3 above)
Comment (svillafe) [inline assets/helpers/abtest.js]: yes, this new annotation was introduced in es6 as a new way of defined [computed property names](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Object_initializer#Computed_property_names)

**Comment by svillafe** [comment]
It is working on **PROD**.

**Pull request merged**